src/KdbOptions.h

```diff
@@ -709,7 +659,7 @@ inline void KdbOptions::HandleNullMapping<arrow::Type::FIXED_SIZE_BINARY>( const
 template<>
 inline void KdbOptions::HandleNullMapping<arrow::Type::DATE32>( const std::string& key, K value )
 {
-  if( -KI == value->t ){
+  if( value->t == -KI || value->t == -KD ){

```

Comment:
I don't think you need to allow -KI here

Reply:
Yep, looks sub-optimal